src/TestProblem/Hydro/Zeldovich/Init_TestProb_Zeldovich.cpp

```diff
@@ -0,0 +1,537 @@
+//
+// Return      :  None
+//-------------------------------------------------------------------------------------------------------
+void SetParameter()
+{
+
+   if ( MPI_Rank == 0 )    Aux_Message( stdout, "   Setting runtime parameters ...\n" );
+
+
+// (1) load the problem-specific runtime parameters
+   const char FileName[] = "Input__TestProb";
+   ReadPara_t *ReadPara  = new ReadPara_t;
+
+// (1-1) add parameters in the following format:
+// --> note that VARIABLE, DEFAULT, MIN, and MAX must have the same data type
+// --> some handy constants (e.g., Useless_bool, Eps_double, NoMin_int, ...) are defined in "include/ReadPara.h"
+// ********************************************************************************************************************************
+// ReadPara->Add( "KEY_IN_THE_FILE",   &VARIABLE,               DEFAULT,      MIN,              MAX               );
+// ********************************************************************************************************************************
+   ReadPara->Add( "Gas_Par_Setup",     &Gas_Par_Setup,          1,            1,                2                 );
+   ReadPara->Add( "n_Pert_Wave_Len",   &n_Pert_Wave_Len,        2,            1,                NoMax_int         );
+   ReadPara->Add( "zc_Collapse",       &zc_Collapse,            1.0,          0.0,              NoMax_double      );
+   ReadPara->Add( "GasTemp_Init",      &GasTemp_Init,           100.0,        0.0,              NoMax_double      );
+   ReadPara->Add( "NPar_X",            &NPar_X,                 NX0_TOT[0],    0,                NoMax_int         );
+
+   ReadPara->Read( FileName );
+
+   delete ReadPara;
+
+
+// (1-2) set the default values
+
+// (1-3) check the runtime parameters
+
+
+// (2) set the problem-specific derived parameters
+#  ifdef COMOVING
+   H0            = HUBBLE0*(100*Const_km/(Const_s*Const_Mpc))*UNIT_T;
+   Pert_Wave_Len = amr->BoxSize[0]/n_Pert_Wave_Len;
+   rho_crit_i    = 1;             // [UNIT_D]; comoving critical density = 3H0^2/8*pi*G
+   k_Pert        = 2.0*M_PI/Pert_Wave_Len;
+   dhx           = amr->BoxSize[0] / NPar_X;
+   NPar_YZ       = amr->BoxSize[1] / dhx;
+#  endif
+
+
+// (3) reset other general-purpose parameters
+//     --> a helper macro PRINT_WARNING is defined in TestProb.h
+   const long   End_Step_Default = __INT_MAX__;
+   const double End_T_Default    = 1.0;
+
+   if ( END_STEP < 0 ) {
+      END_STEP = End_Step_Default;
+      PRINT_WARNING( "END_STEP", END_STEP, FORMAT_LONG );
+   }
+
+   if ( END_T < 0.0 ) {
+      END_T = End_T_Default;
+      PRINT_WARNING( "END_T", END_T, FORMAT_REAL );
+   }
+
+#  ifdef PARTICLE
+   if ( Gas_Par_Setup == 1 ) // overwrite the total number of particles in gas-only setup
+   {
+      amr->Par->NPar_Active_AllRank = 0;
+   }
+   if ( Gas_Par_Setup == 2 ) // overwrite the total number of particles in particle-only setup
+   {
+      amr->Par->NPar_Active_AllRank = NPar_X*SQR(NPar_YZ);
+   }
+#  endif
```

Comment:
Add `PRINT_WARNING( "PAR_NPAR", amr->Par->NPar_Active_AllRank, FORMAT_LONG );` before this line.